CMake/folly-deps.cmake

```diff
@@ -26,7 +26,11 @@ find_package(gflags CONFIG QUIET)
 if (gflags_FOUND)
   message(STATUS "Found gflags from package config")
   set(FOLLY_HAVE_LIBGFLAGS ON)
-  set(FOLLY_SHINY_DEPENDENCIES ${FOLLY_SHINY_DEPENDENCIES} gflags)
+  if(EXISTS "/etc/fedora-release")
+    set(FOLLY_SHINY_DEPENDENCIES ${FOLLY_SHINY_DEPENDENCIES} gflags-shared)
```

Comment:
Is it called `gflags-shared` for every Fedora version or only after some threshold?

Reply:
I've checked latest gflags packages for:
Fedora Core 27: gflags-2.1.2-5
Fedora Core 26: gflags-2.1.2-2
Fedora Core 25: gflags-2.1.2-1
In all of them it's called `gflags-shared`.

I haven't checked previous versions of packages or distributions. Current stable is FC27. Hope this helps. Thank you!

Reply:
`gflags-shared` is one of the targets GFlags can choose to export by default. This check can be done as 
```CMake
if (TARGET gflags-shared)
  set(FOLLY_SHINY_DEPENDENCIES ${FOLLY_SHINY_DEPENDENCIES} gflags-shared)
elseif (TARGET gflags)
  set(FOLLY_SHINY_DEPENDENCIES ${FOLLY_SHINY_DEPENDENCIES} gflags)
else()
  message(FATAL_ERROR "Unable to determine the target name for the GFlags package.")
endif()
```

Reply:
Awesome! That kind of check was something that I was actually looking for.

Reply:
Please do that.